frontend/src/components/pages/volunteer/posting/VolunteerPostingAvailabilities.tsx

```diff
@@ -54,17 +75,58 @@ const POSTING = gql`
   }
 `;
 
+// WE pass the user id from our current session token
+// TODO: Make this pass in a array of DTO
+const SUBMIT_SIGNUPS = gql`
+  mutation VolunteerPostingAvailabilities_SubmitSignups(
+    $upsertDeleteShifts: UpsertDeleteShiftSignupRequestDTO!
+  ) {
+    upsertDeleteShiftSignups(upsertDeleteShifts: $upsertDeleteShifts) {
+      status
+    }
+  }
+`;
+
 const VolunteerPostingAvailabilities = (): React.ReactElement => {
+  const currentUser: AuthenticatedUser = getLocalStorageObj<AuthenticatedUser>(
+    AUTHENTICATED_USER_KEY,
+  );
+  const [shiftSignups, setShiftSignups] = useState<SignupRequest[]>([]);
+  const [deleteSignups, setDeleteSignups] = useState<DeleteSignupRequest[]>([]);
+
   const { id } = useParams<{ id: string }>();
   const {
     loading: isShiftsLoading,
-    error,
-    data: { shiftsByPosting } = {},
-  } = useQuery(SHIFTS_BY_POSTING, {
-    variables: { postingId: id },
+    error: shiftError,
+    data: { shiftsWithSignupsAndVolunteersByPosting: shiftsByPosting } = {},
+  } = useQuery(SHIFTS_WITH_SIGNUPS_BY_POSTING, {
+    variables: { postingId: id, userId: currentUser?.id },
     fetchPolicy: "cache-and-network",
+    onCompleted: () => {
+      setShiftSignups(
+        ((shiftsByPosting as ShiftWithSignupAndVolunteerResponseDTO[]) ?? [])
+          .filter((shift) => shift.signups.length > 0)
+          .map((shift) => {
+            return {
+              shiftId: shift.id,
+              note: shift.signups[0].note,
+            };
+          }),
+      );
+      setDeleteSignups(
+        ((shiftsByPosting as ShiftWithSignupAndVolunteerResponseDTO[]) ?? [])
+          .filter((shift) => shift.signups.length > 0)
+          .map((shift) => {
+            return {
+              shiftId: shift.id,
+              toDelete: false,
+            };
+          }),
+      );
```

Comment:
```suggestion
      const initDeleteSignups = ((shiftsByPosting as ShiftWithSignupAndVolunteerResponseDTO[]) ?? [])
          .filter((shift) => shift.signups.length > 0)
          .map((shift) => {
            return {
              shiftId: shift.id,
              toDelete: false,
            };
          }),
          
      setDeleteSignups(initDeleteSignups);
```

IMO it's easier to read when you're not passing a function into a function

Reply:
fair, will make the change : )